assets/src/components/AssignmentTable.js

```diff
@@ -164,21 +169,29 @@ function AssignmentTable (props) {
 
   // this effect scrolls to current week of assignments if it exists
   useEffect(() => {
-    if (currentWeekRow.current) {
-      const tableHeaderOffset = 120
+    if (previousWeekRow.current) {
+      const tableHeaderOffset = -74 // Manually measured height of table header row
       tableRef.current.scrollTo({
-        top: currentWeekRow.current.offsetTop - tableHeaderOffset,
+        top: previousWeekRow.current.offsetTop - tableHeaderOffset,
         behavior: 'smooth'
       })
     }
-  }, [currentWeekRow.current])
+  }, [previousWeekRow.current])
 
   useEffect(() => {
     const allGroupNames = assignments.map(ag => ag.assignmentGroup.name)
     const uniqueGroupNames = [...new Set(allGroupNames)].sort()
     setAssignmentGroupNames(uniqueGroupNames)
   }, [assignments])
 
+  useEffect(() => {
+    setWeeksPresentInCollection([...new Set(assignments.map(a => a.week))].sort((a, b) => { return a - b }))
+    const indexOfCurrentWeek = weeksPresent.indexOf(currentWeek)
+    if (indexOfCurrentWeek > 0) {
+      setPreviousWeek(weeksPresent[indexOfCurrentWeek - 1])
+    }
+  }, [assignments])
```

Comment:
when user set the course goal from previous, in that case the scroll is not happening at all. I tried to remove it and tried various option but still scroll don't work. Some how when course goal is set ( from before ) assignment is [] when this effect takes place

Reply:
@chrisrrowland could you explain why you're setting `weeksPresent.current` here? I'm actually not exactly sure what this block of code is doing in general - could you explain it at a high level? Thanks!

Reply:
I believe @pushyamig added some of that, but the underlying concept was that not all calendar weeks necessarily have any assignments, so it may go 
Week 1
Week 2
Week 4
Week 5
etc

And when working off of the previous week, we just wanted it to be the previous week with assignments, so if it was currently Week 5 we'd scroll relative to Week 4, and if it were Week 4 we'd scroll relative to week 2.

Reply:
Ah, I see, thanks for clarifying.

Reply:
For some reason when user never sets a course goal ( with filtered assignment in place) the `AssignmentTable` lacks an exact render so the assignment weeks value is available in between the render and not on the final render. So I have store all the week list as a local variable `weeksPresent`.